core/src/main/java/com/opensymphony/xwork2/util/fs/StrutsJarURLConnection.java

```diff
@@ -20,34 +20,89 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
 import java.net.JarURLConnection;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLConnection;
+import java.net.URLDecoder;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardCopyOption;
 import java.security.AccessController;
 import java.security.PrivilegedActionException;
 import java.security.PrivilegedExceptionAction;
+import java.util.jar.JarEntry;
 import java.util.jar.JarFile;
 
 /**
- * WW-4901 Decouples from underlying implementation of {@link URL#openConnection()}
+ * WW-4901 If was needed, decouples from underlying implementation of {@link URL#openConnection()}
  * e.g. from IBM WebSphere com.ibm.ws.classloader.Handler$ClassLoaderURLConnection
+ * WW-4920 Also decouples from and fixes {@link JarURLConnection#parseSpecs(URL)} if was needed
+ * e.g. from Oracle WebLogic which may report jar urls like "zip:C:/web-app-lib-path/some-jar.jar"
+ * but {@link JarURLConnection#parseSpecs(URL)} breaks on such urls
+ * While {@link JarURLConnection#parseSpecs(URL)} is private, then we had to extend {@link URLConnection} instead
  * @since 2.5.15
```

Comment:
`2.5.16` or just remove it.

Reply:
Thanks! Did you mean only here which is modified in 2.5.16? Or all "2.5.15" occurrences? Because there are also other "2.5.15" occurrences in other .java files e.g. BackgoundProcess.java.

Reply:
We can leave it, there is a tag for 2.5.15 and it didn't just pass the verification process to be a full release.

Reply:
Ah, didn't noticed that this class was already there in 2.5.15. Then it is ok.